### DISCUSSION TITLE
[Bug]: ERROR in ./node_modules/pdfjs-dist/build/pdf.mjs 402:14 Loader files .mjs

### DISCUSSION THREAD

**Discussion by GuiSandoval** [mozilla/pdf.js#18498] [Q&A]
### Attach (recommended) or Link to PDF file

In the new version of pdfjs-dist (4.4.168), the project is not running in React.
Previously, I used version 3.6.172, and it worked, but I need the updated library.

### Web browser and its version

React 16.9.0 / node v12.18.4 / webpack 4.41.2 / babel-loader 8.1.0

### Operating system and its version

linux/windows

### PDF.js version

4.4.168

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

1 - install pdfjs-dist 
2 - execute command
`webpack-dev-server --watch --port 4000 --progress --colors --open --mode development`

### What is the expected behavior?

project running

### What went wrong?

![image](https://github.com/user-attachments/assets/a0300c29-7bee-4a73-b2a0-061359bac36e)

```
ERROR in ./node_modules/pdfjs-dist/build/pdf.mjs 402:14
Module parse failed: Unexpected token (402:14)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| }
| function _isValidProtocol(url) {
>   switch (url?.protocol) {
|     case "http:":
|     case "https:":
 @ ./node_modules/react-pdf/dist/esm/index.js 1:0-36 12:0-25 13:0-128
 @ ./src/main/components/shared/pdfViewer/pdfViewer.tsx
 @ ./src/main/components/layouts/layoutMain.tsx
 @ ./src/main/routes/privateRoute.tsx
 @ ./src/main/routes/index.ts
 @ ./src/main/components/routes/routes.tsx
 @ ./src/main/app.tsx
 @ ./src/index.tsx
```

### Additional context

The solution that almost worked was adding the Babel loader for .mjs files. The project runs, but when accessing the site in the browser, the following error appears:

![image](https://github.com/user-attachments/assets/1de9a083-b3cf-40ec-8b40-e003af827d51)

![image](https://github.com/user-attachments/assets/405e47a4-385b-45ee-9d20-97f1309bc1e5)

**Comment by yanglan-0726** [comment]
Is there any solution to this?

**Comment by GuiSandoval** [comment]
I managed to solve the problem by updating my webpack version to 5.93.0.
This way, it worked for me.
![image](https://github.com/user-attachments/assets/e7009ed1-41be-4049-94a4-060f908adbf1)

**Comment by mayconghizoni** [comment]
Any other solution for this problem? It is not feasible to update the webpack version here

**Reply by leon-epam** [reply]
I also have the error " Cannot read properties of undefined (reading 'd')", and my webpack version is webpack 4 too,it's a big problem😭.

**Comment by tovafrishman** [comment]
maybe its because node version , try to upgrade to >16
